packages/cozy-doctypes/src/contacts/Contact.js

```diff
@@ -0,0 +1,37 @@
+const getPrimaryOrFirst = property => obj => {
+  if (!obj[property] || obj[property].length === 0) return ''
+  return obj[property].find(property => property.primary) || obj[property][0]
+}
+
+class Contact extends Document {
+  static getInitials(name) {
+    if (!name) return ''
+    if (typeof name === 'string') {
+      return name[0].toUpperCase()
+    }
+
+    return ['givenName', 'familyName']
+      .map(part => name[part] || '')
+      .map(name => name[0])
+      .join('')
+      .toUpperCase()
+  }
+
+  // TODO: sadly we have different versions of contacts' doctype to handle...
+  // A migration tool on the stack side is needed here
```

Comment:
Is it still relevant? 

If yes maybe we should add this task to our next migration plan

Reply:
oui, c'est prévu dans la carte Migration !